### PR TITLE
Fixes #30608 - Use new react mounter for templates form

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: app/views/templates/_form.html.erb [hunk 1]
```diff
@@ -40,9 +40,8 @@
       <% end -%>
 
       <div class='form-group'>
-        <div class="col-md-12">
-            <div id="react-ace"></div>
-              <%= mount_react_component('Editor', "#react-ace", {
+        <div id="react-ace" class="col-md-12">
```
Comment (amirfefer) [inline app/views/templates/_form.html.erb]: Can you please move this div to the `Editor` component? 
this id is used in a few integration tests only for this component

**Review by amirfefer — COMMENTED** [review]
Thanks @yifatmakias, tested and works well :+1: 
Just one tiny nit

File: app/views/templates/_form.html.erb (hunk 1 above)
Comment (yifatmakias) [reply app/views/templates/_form.html.erb]: I moved this div to the Editor component. Let me know if what I did is what you meant because I wasn't sure.

**Comment by yifatmakias** [comment]
[test katello]

File: webpack/assets/javascripts/react_app/components/Editor/Editor.js [hunk 2]
```diff
@@ -110,106 +110,108 @@ class Editor extends React.Component {
...
-        />
-        <div
-          id="diff-table"
-          className={selectedView === 'diff' ? '' : 'hidden'}
-        >
-          <DiffView
-            oldText={template || ''}
-            newText={value}
-            viewType={diffViewType}
+      <div id="react-ace">
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/Editor/Editor.js]: this div isn't related to styles but was created for mounting and for testing, so basically we don't need it at all, we can move the related tests to use the div beneath it - `editor-container`

for example: 
```ruby
    # test/integration/ptable_js_test.rb
    test "edit page" do  
      fill_in_editor_field('#editor-container', editor_text) 
      assert has_editor_display?('#editor-container, editor_text')
    end
```

**Comment by yifatmakias** [comment]
[test katello]

**Review by amirfefer — APPROVED** [review]
Thanks @yifatmakias ! works well :+1:

**Pull request merged**